libs/render_delegate/render_delegate.cpp

```diff
@@ -636,6 +636,12 @@ void HdArnoldRenderDelegate::_SetRenderSetting(const TfToken& _key, const VtValu
         return colorManager;
     };
 
+    // When husk/houdini changes frame, they set the new frame number via the render settings.
+    if (_key == str::t_houdiniFrame) {
```

Comment:
I guess we'll receive that render setting even before the first render. Can it be a problem to call Restart() if the render didn't actually start yet ?

Reply:
Good question, we receive some render settings before the first render, the `Restart()` function sets `_needsRestart` which is only used when the render is finished or paused (but it was set before so unlikely to return something else than false). `Restart()` also sets `_paused` to false, which will be used if the render was paused and will call a AiRenderResume. So I am thinking it should be ok, what do you think ?